presto-docs/src/main/sphinx/develop/presto-page.rst

```diff
@@ -0,0 +1,233 @@
+=============   =========================
+BYTE_ARRAY      1
+SHORT_ARRAY     2
+INT_ARRAY       4
+LONG_ARRAY      8
+INT128_ARRAY    16
+=============   =========================
+
+Let’s take the example from the Null Flags section and say that we have an
+integer column with 10 rows with nulls in zero-based rows 1, 4, 6, 7, 9.
+We’ll have 4 bytes storing the number of rows: 10, followed by 3 bytes of
+null flags, followed by 20 bytes representing 5 non-null integer values for
+rows 0, 2, 3, 5, 8..
+
+.. image:: presto-page-int-column.png
+  :width: 500
+
+VARIABLE_WIDTH Encoding
+-----------------------
+
+* Number of rows - 4 bytes
+* Offsets - number of rows * 4 bytes; 4 bytes per offset
+* Null flags
+* Total number of bytes in all values - 4 bytes
+* Concatenated values
+
+Let’s again take the example from the Null Flags section and say that we
+have a string column with 10 rows with nulls in zero-based rows 1, 4, 6, 7, 9.
+The non-null rows will have values: 0 - Denali, 2 - Reinier, 3 - Whitney,
+5 - Bona, 8 - Bear. We’ll have 4 bytes storing the number of rows: 10, followed
+by 40 bytes of offsets, followed by 3 bytes of null flags, followed by 1 bytes
+storing total size of all strings: 28, followed by the concatenated string values.
+Notice that we have offsets for all rows, not just the non-null rows.
+
+.. image:: presto-page-string-column.png
+  :width: 800
+
+ARRAY Encoding
+--------------
+
+* Elements column
+* Number of rows - 4 bytes
+* Offsets - (number of rows + 1) * 4 bytes; 4 bytes per offset
+* Null flags
+
+An array column with 10 rows is represented like this:
+
+.. image:: presto-page-array-column.png
+  :width: 400
+
+MAP Encoding
+------------
+
+* Keys column
+* Values column
+* Hash table size - 4 bytes
+* [optional] Hash table
+* Number of rows - 4 bytes
+* Offsets - (number of rows + 1) * 4 bytes; 4 bytes per offset
+* Null flags
+
+A map column with 10 rows is represented like this:
+
+.. image:: presto-page-map-column.png
+  :width: 600
+
+ROW Encoding
+------------
+
+* Number of fields - 4 bytes
+* One column per field
+* Number of rows - 4 bytes
+* Offsets - (number of rows + 1) * 4 bytes; 4 bytes per offset
+* Null flags
+
+Nested columns are serialized only for non-null rows. In the presence of
+null rows, the row numbers of the nested columns don’t match the top-level
+row numbers. Offsets are specifying the row numbers of the nested columns.
+
+Let’s again take the example from the Null Flags section and say that we
+have a column of type ROW(a, b, c, d) with 10 rows with nulls in zero-based
+rows 1, 4, 6, 7, 9. Nested columns will have only 5 rows and offsets will
+be: 0, 0, 1, 2, 0, 3, 0, 0, 4, 0. Offsets for null rows are zero.
+
+Note: Offsets are redundant information as they can be reconstructed
+from the null flags.
+
+.. image:: presto-page-row-column.png
+  :width: 600
+
+Additional Usage
+----------------
+
+SerializedPage format is also used to specify constant values in the plan
```

Comment:
what is preferred name for future reference? If it is PrestoPage format,  consider to use PrestoPage instead of SerializedPage format here

Reply:
Let's keep SerializedPage to avoid introducing new term. I'll make the change.

Reply:
Yeah, let's keep SerializedPage